docs/deployment-ux.md

```diff
@@ -0,0 +1,410 @@
+Blox Deployments User Experience
+================================
+
+Use Cases
+---------
+
+The following are typical use cases for Deployments:
+
+-   Create an Environment to deploy a new environment revision. The
```

Comment:
We discussed terminology around this a bit. In particular, we have some overlapping concepts with k8s that have different meanings there (particularly `deployment`). We're still evaluating whether we should adopt the terms that are already in widespread use with k8s (and run the risk of confusion because they don't behave exactly the same), or to use entirely new terms (and make it harder for new customers familiar with container orchestration to understand the entities in our system). Having duplicate terms with totally different meanings seems like something we should try and avoid, though.